fieldsolver/ldz_electric_field.cpp

```diff
@@ -1639,105 +1886,100 @@ void calculateElectricField(
-   cint& RKCase
-) {
-   //const std::array<int, 3> gridDims = technicalGrid.getLocalSize();
+    FsGrid<std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBGrid,
+    FsGrid<std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH>& perBDt2Grid,
+    FsGrid<std::array<Real, fsgrids::efield::N_EFIELD>, FS_STENCIL_WIDTH>& EGrid,
+    FsGrid<std::array<Real, fsgrids::efield::N_EFIELD>, FS_STENCIL_WIDTH>& EDt2Grid,
+    FsGrid<std::array<Real, fsgrids::ehall::N_EHALL>, FS_STENCIL_WIDTH>& EHallGrid,
+    FsGrid<std::array<Real, fsgrids::egradpe::N_EGRADPE>, FS_STENCIL_WIDTH>& EGradPeGrid,
+    FsGrid<std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH>& momentsGrid,
+    FsGrid<std::array<Real, fsgrids::moments::N_MOMENTS>, FS_STENCIL_WIDTH>& momentsDt2Grid,
+    FsGrid<std::array<Real, fsgrids::dperb::N_DPERB>, FS_STENCIL_WIDTH>& dPerBGrid,
+    FsGrid<std::array<Real, fsgrids::dmoments::N_DMOMENTS>, FS_STENCIL_WIDTH>& dMomentsGrid,
+    FsGrid<std::array<Real, fsgrids::bgbfield::N_BGB>, FS_STENCIL_WIDTH>& BgBGrid,
+    FsGrid<fsgrids::technical, FS_STENCIL_WIDTH>& technicalGrid,
+    SysBoundary& sysBoundaries,
+    cint& RKCase) {
+   // const std::array<int, 3> gridDims = technicalGrid.getLocalSize();
    const int* gridDims = &technicalGrid.getLocalSize()[0];
-   const size_t N_cells = gridDims[0]*gridDims[1]*gridDims[2];
-   phiprof::Timer upwindedETimer {"Calculate upwinded electric field"};
-   int computeTimerID {phiprof::initializeTimer("Electric field compute cells")};
-   
-   phiprof::Timer mpiTimer {"Electric field ghost updates MPI", {"MPI"}};
+   const size_t N_cells = gridDims[0] * gridDims[1] * gridDims[2];
+   phiprof::Timer upwindedETimer{"Calculate upwinded electric field"};
+   int computeTimerID{phiprof::initializeTimer("Electric field compute cells")};
+
+   phiprof::Timer mpiTimer{"Electric field ghost updates MPI", {"MPI"}};
    // Update ghosts if necessary, unless previous terms have already updated them
-   if(P::ohmHallTerm > 0) {
+   if (P::ohmHallTerm > 0) {
       EHallGrid.updateGhostCells();
    }
-   if(P::ohmGradPeTerm > 0) {
+   if (P::ohmGradPeTerm > 0) {
       EGradPeGrid.updateGhostCells();
    }
-   if(P::ohmHallTerm == 0) {
+   if (P::ohmHallTerm == 0) {
       dPerBGrid.updateGhostCells();
    }
-   if(P::ohmHallTerm == 0 && P::ohmGradPeTerm == 0) {
+   if (P::ohmHallTerm == 0 && P::ohmGradPeTerm == 0) {
       dMomentsGrid.updateGhostCells();
    }
-   
+
    mpiTimer.stop();
-   
-   // Calculate upwinded electric field on inner cells
-   #pragma omp parallel
+
+// Calculate upwinded electric field on inner cells
+#pragma omp parallel
    {
-      phiprof::Timer computeTimer {computeTimerID};
-      #pragma omp for collapse(2)
-      for (int k=0; k<gridDims[2]; k++) {
-         for (int j=0; j<gridDims[1]; j++) {
-            for (int i=0; i<gridDims[0]; i++) {
+      phiprof::Timer computeTimer{computeTimerID};
+#pragma omp for collapse(2)
```

Comment:
No you said openmp, so you don't like it pushing all pragmas to the beginning of the line?

Reply:
Exactly. I much prefer them to be aligned with the loop they apply to.